src/python/pants/backend/go/util_rules/sdk.py

```diff
@@ -58,49 +60,57 @@ def __init__(
 @dataclass(frozen=True)
 class GoSdkRunSetup:
     digest: Digest
-    script: FileContent
+    path: str
 
     CHDIR_ENV = "__PANTS_CHDIR_TO"
 
 
 @rule
 async def go_sdk_invoke_setup(goroot: GoRoot) -> GoSdkRunSetup:
-    # Note: The `go` tool requires GOPATH to be an absolute path which can only be resolved
-    # from within the execution sandbox. Thus, this code uses a bash script to be able to resolve
-    # absolute paths inside the sandbox.
-    go_run_script = FileContent(
-        "__run_go.sh",
+    # Note: The `go` tool requires GOPATH to be an absolute path, which can only be resolved
+    # from within the execution sandbox. Thus, this code uses a script to resolve absolute paths.
+    script = FileContent(
+        "__run_go.py",
         textwrap.dedent(
             f"""\
-            export GOROOT={goroot.path}
-            export GOPATH="$(/bin/pwd)/gopath"
-            export GOCACHE="$(/bin/pwd)/cache"
-            /bin/mkdir -p "$GOPATH" "$GOCACHE"
-            if [ -n "${GoSdkRunSetup.CHDIR_ENV}" ]; then
-              cd "${GoSdkRunSetup.CHDIR_ENV}"
-            fi
-            exec "{goroot.path}/bin/go" "$@"
+            import os, sys
+
+            cwd = os.getcwd()
+            gopath = os.path.join(cwd, "gopath")
+            gocache = os.path.join(cwd, "cache")
+
+            os.environ["GOROOT"] = "{goroot.path}"
+            os.environ["GOPATH"] = os.path.join(cwd, "gopath")
+            os.environ["GOCACHE"] = os.path.join(cwd, "cache")
+
+            os.makedirs(gopath, exist_ok=True)
+            os.makedirs(gocache, exist_ok=True)
+
+            chdir = os.environ["{GoSdkRunSetup.CHDIR_ENV}"]
+            if chdir:
+                os.chdir(chdir)
+
+            os.execv(os.path.join("{goroot.path}", "bin", "go"), ["go", *sys.argv[1:]])
```

Comment:
Note that Python uses the first arg in the list to set `argv[0]`, i.e. the program name.